inc/form.class.php

```diff
@@ -2402,31 +2398,33 @@ public function validationRequired(): bool {
     * @return boolean true if the user can use the form
     */
    public function canViewForRequest(): bool {
-      global $DB;
-
       if ($this->isNewItem()) {
          return false;
       }
 
-      if ($this->fields['access_rights'] != self::ACCESS_PUBLIC) {
-         if (Session::getLoginUserID() === false || !$this->checkEntity(true)) {
-            return false;
-         }
+      // Public forms -> always visible
+      if ($this->fields['access_rights'] == self::ACCESS_PUBLIC) {
+         return true;
       }
 
-      if (!Session::haveRight('entity', UPDATE) && $this->fields['access_rights'] == self::ACCESS_RESTRICTED) {
-         $iterator = $DB->request(PluginFormcreatorForm_Profile::getTable(), [
-            'WHERE' => [
-               'profiles_id'                 => $_SESSION['glpiactiveprofile']['id'],
-               'plugin_formcreator_forms_id' => $this->getID()
-            ],
-            'LIMIT' => 1
-         ]);
-         if (count($iterator) == 0) {
-            return false;
-         }
+      // Restricted and private forms -> Check session
+      if (Session::getLoginUserID() === false || !$this->checkEntity(true)) {
+         return false;
+      }
+
+      // Super admin can always access any forms
```

Comment:
update right on entity is not reserved to super admins. I would change this comment
```suggestion
      // form administrators can always access any forms
```

Reply:
Done in a3198897.